### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by klucar**
Completely new dashboard single page application for Myriad. 

Uses Facebook's React component system to build and update user interface completely on the client side.

Uses npm for dependencies and gulp to compile js artifacts together.

### CODE REVIEW

**Comment by adam-mesos** [comment]
Is this apache-licensed, or friendly with the Apache license?

**Comment by klucar** [comment]
From what I can tell, yes it is Apache compatible. 

I ran the nlf (https://www.npmjs.com/package/nlf) tool on the package.json. This traverses the entire dependency tree for license information. It found primarily BSD and MIT. There were some ISC (http://en.wikipedia.org/wiki/ISC_license) that I couldn't find a direct reference to on the Apache site, but it looks open to me. There were a handful of modules that didn't specify a license, and one WTFPL (http://www.wtfpl.net/) which seems super compatible.

**Comment by klucar** [comment]
I updated to the latest phase1 branch and resolved the conflicts. @DarinJ said he'll take a look and verify it runs soon.

File: setup-yarn-2.sh [hunk 1]
```diff
@@ -62,3 +63,4 @@ sudo -u hduser sh -c '/usr/local/hadoop/sbin/mr-jobhistory-daemon.sh start histo
 
 # Check status
 sudo -u hduser -u hduser jps
+
```
Comment (DarinJ) [inline setup-yarn-2.sh]: Why are we modifying this file? seems irrelevant to the dashboard.  If it needs fixed should it be a separate PR?

File: src/main/java/com/ebay/myriad/api/ClustersResource.java [hunk 2]
```diff
@@ -68,8 +67,15 @@ public Response flexUp(FlexUpClusterRequest request) {
                 "request object cannot be null or empty");
 
         // TODO(mohit): Validation
+        LOGGER.info("Received Flexup Cluster Request");
+
+
```
Comment (DarinJ) [inline src/main/java/com/ebay/myriad/api/ClustersResource.java:72]: Just one new line here.

File: src/main/resources/webapp/js/components/ClustersComponent.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  displayName: "ClustersComponent",
+
+  render: function () {
+    return(
+      <h1>Clusters</h1>
+   )}
+
+});
+
+module.exports = ClustersComponent;
```
Comment (DarinJ) [inline src/main/resources/webapp/js/components/ClustersComponent.js]: I don't really get the purpose of the clusters page.  Is this something you need for future work?

**Comment by DarinJ** [comment]
A few in-line comments.  It would be nice to have a line in one of the main documentation pages mention the WebUI.  In a shippable state, should be merged.

**Comment by DarinJ** [comment]
@kensipe were you going to look at this?  It's a blocker for a few other issues.

**Comment by kensipe** [comment]
@DarinJ I fell behind on some work priorities... I'm on it now.   Did your comments get resolved?   specifically the comment about unrelated changes.  Has that been pulled out or will it be?

**Comment by DarinJ** [comment]
@kensipe, Jim hasn't fixed the unrelated changes (setup-yarn-2.sh) though looking at the file I'd say its in a better state, but he can pull that out easy enough.

**Comment by klucar** [comment]
just removed changes to setup-yarn-2.sh. Squashed all working commits into
one. Reflected issue #82 in commit message. pull request updated.

File: src/main/resources/webapp/js/components/ClustersComponent.js (hunk 3 above)
Comment (kensipe) [inline src/main/resources/webapp/js/components/ClustersComponent.js]: I agree with @DarinJ.  I think I understand where this is going, however it is a potential future.   Currently we only support 1 cluster per framework.   unless I'm missing something... lets remove.

File: src/main/resources/webapp/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+The gulpfile contains a dev target that launches a node.js webserver and watches the webapp files, re-assembling when
+files change. To launch simply run
+
+```
+gulp dev
+```
+
+A browser window should open with the site loaded. If not, it uses [port 8888](http://localhost:8888) 
+It is helpful to have myriad setup in Vagrant locally so the api is available.
+
```
Comment (kensipe) [inline src/main/resources/webapp/README.md]: I'm fine with this information being here... however it is an important part of the build process and a new developer should not have to hunt for this information.  there needs to be a comment and a link from the main project README to this file.

File: src/main/resources/webapp/js/components/ConfigComponent.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+var React = require('react');
+var ProfileComponent = require('../components/ProfileComponent');
+var RawJSONComponent = require('../components/RawJSONComponent');
+
+// localhost:8192/api/config
+//{"profiles":{"small":{"cpu":"1","mem":"1100"},"medium":{"cpu":"2","mem":"2048"},"large":{"cpu":"4","mem":"4096"}},"mesosMaster":"10.0.2.15:5050"}
```
Comment (kensipe) [inline src/main/resources/webapp/js/components/ConfigComponent.js]: why is this commented out?   can we just remove this?

File: src/main/resources/webapp/js/components/Myriad.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+/*
+  shouldComponentUpdate: function() {
+
+  },
+
+  componentWillUnmount: function() {
+
+  },
+*/
```
Comment (kensipe) [inline src/main/resources/webapp/js/components/Myriad.js]: remove commented out code

File: src/main/resources/webapp/js/components/ProfileComponent.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+var React = require('react');
+
+//{"small":{"cpu":"1","mem":"1100"}}
```
Comment (kensipe) [inline src/main/resources/webapp/js/components/ProfileComponent.js]: lets remove commented out code

File: src/main/resources/webapp/js/components/TaskListComponent.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+var React = require('react');
+
+var TaskListComponent = React.createClass({
+  displayName: "TaskListComponent",
+
+  render: function () {
+
+    // props: name, taskNames
```
Comment (kensipe) [inline src/main/resources/webapp/js/components/TaskListComponent.js]: comment isn't meaningful... I don't see any value to the comment

**Comment by kensipe** [comment]
@klucar I've made a few comments... lets get this merged today if possible.   Here is what is outstanding:
1. the setup-yarn-2.sh changes should be removed from this PR.  you reference in your last comment that this is taken care of, but I don't see it.
2. remove cluster stuff
3. add link to webapp/README in the main README (with comment)
4. remove commented out code from ConfigComponent.js, Myriad.js, ProfileComponent.js
   commented code:  TaskListComponent.js
5. misalignment of input text boxes on the flex tab

**Comment by klucar** [comment]
The Clusters page is there only because the original UI had one. I couldn't
figure out what the original actually displayed, so this one is blank. I'll
gladly remove it, unless someone knows what it was meant for.

Most of the commented out stuff is just my development cruft so I can
remember what JSON the web service returns. I'll clean it up a bit, but the
props stuff is useful, so I may just clarify what they are better.

I should be able to do this later today.

Jim

**Comment by kensipe** [comment]
@klucar are you expecting this today?  I understand this is blocking some other work.

**Comment by klucar** [comment]
I'm building and testing now.

How should I do the README link? (e.g. relative path, absolute github path,
something else)

**Comment by kensipe** [comment]
I would use a relative link such that it works from github.

**Comment by klucar** [comment]
what do you think about just moving it to the docs directory with
everything else? this makes more sense to me.

**Comment by klucar** [comment]
I moved the README.md to docs/myriad-dashboard.md

**Comment by klucar** [comment]
Latest code should be good to go. Not sure why my setup-yarn-2.sh fix didn't take last time, but it looks like latest doesn't have any changes to setup-yarn-2.sh

**Comment by kensipe** [comment]
LGTM!   Merging!
